### PR TITLE
Fix.  Temperature fan min speed should be used

### PR DESCRIPTION
**Pull request by wizhippo**
min_speed should be the minimum speed.  at the moment if below min_speed the speed will be set to 0 instead of the specified min.

Signed-off-by: Douglas Hammond wizhippo@gmail.com

### CODE REVIEW

**Comment by KevinOConnor** [comment]
Thanks.  If the requested value is zero, shouldn't it turn off the fan?

-Kevin

**Comment by wizhippo** [comment]
Maybe?  If the min speed is set should the fan be allowed to go below this value?  I guess it is needed for bang bang mode.

I have added speed 0 support.

File: klippy/extras/temperature_fan.py [hunk 1]
```diff
@@ -114,7 +116,7 @@ def temperature_callback(self, read_time, temp):
         co = self.Kp*temp_err + self.Ki*temp_integ - self.Kd*temp_deriv
         bounded_co = max(0., min(self.temperature_fan.max_speed, co))
         self.temperature_fan.set_speed(
-            read_time, self.temperature_fan.max_speed - bounded_co)
+            read_time, max(self.min_speed, self.temperature_fan.max_speed - bounded_co))
```
Comment (wizhippo) [inline klippy/extras/temperature_fan.py]: not sure if bounded_co could instead take min_speed into account or not.

**Comment by KevinOConnor** [comment]
It looks like this PR doesn't currently pass the build tests.  Also, looks like the most recent additions peek into the fan module's member variables - that should be avoided.

-Kevin

**Comment by KevinOConnor** [comment]
If I understand the code correctly, the temperature_fan will turn off completely if the measured temperature is notably below the target temperature.  Wouldn't this change effectively leave the fan always running?  Is that desirable?

-Kevin

**Comment by wizhippo** [comment]
That is the intention, to always have the fan at a minimal speed.

I currently use this to have my case always cooled at some level. Then I have a thermistor on my x stepper heat sync.  When it heats up I wish my case fan speed to increase from the minimal speed when needed.

As other components in my case need cooling too and the thermistor is on the stepper driver I though this would be a good compromise from adding more thermistor as i do not have more adc inputs.

**Comment by KevinOConnor** [comment]
Thanks.  I went to commit this, but I don't see a signed-off-by line (as described at: https://github.com/KevinOConnor/klipper/blob/master/docs/CONTRIBUTING.md ).

BTW, in regard to "peeking at member variables", I was referring to temperature_fan.py code looking into member variables of fan.py code.  I appologise if I made an incorrect review observation before.  The way you have the code now is fine too, of course.

Also, it looks like we'll need to update docs/Config_Changes.md to note the difference in interpretation of fan_controller.min_speed.

-Kevin

**Comment by KevinOConnor** [comment]
Thanks.

-Kevin

**Pull request merged**